### PR TITLE
INT-4233: Prework - Simple JSON Serializer

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-4233

### CODE REVIEW

**Comment by garyrussell** [comment]
I am going to change it to a utility and make it stateless.

**Comment by garyrussell** [comment]
Should be good to go now.

File: spring-integration-core/src/main/java/org/springframework/integration/json/SimpleJsonSerializer.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			catch (InvocationTargetException e) {
+				if (logger.isDebugEnabled()) {
+					logger.debug("Failed to serialize property " + method.getName(), e);
+				}
+				result = e.getMessage();
+			}
+			stringBuilder.append(toElement(result)).append(",");
+		}, method -> {
+			String name = method.getName();
+			return (name.length() > 3 && name.startsWith("get") && propertyMap.keySet().contains(name.substring(3)))
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/json/SimpleJsonSerializer.java]: Can't we just use `BeanUtils.getPropertyDescriptors()`?

OTOH wouldn't it be better to introspect all the properties unless `ignoreProperties` provided instead?
For example the `FileInfo` should be serialized fully except `getFileInfo()` result. 
Isn't it?

File: spring-integration-core/src/test/java/org/springframework/integration/json/SimpleJsonSerializerTests.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		Foo foo = new Foo();
+		String json = SimpleJsonSerializer.toJson(foo, "foo", "bar", "dub", "bool");
+		assertThat(json, startsWith("{"));
+		assertThat(json, containsString("\"bool\":true"));
+		assertThat(json, containsString("\"bar\":42"));
+		assertThat(json, containsString("\"foo\":\"bar\""));
+		assertThat(json, containsString("\"dub\":1.6"));
+		assertThat(json, endsWith("}"));
+		String[] split = json.split(",");
+		assertThat(split, arrayWithSize(4));
```
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/json/SimpleJsonSerializerTests.java]: ??? Why don't test that it is fully compatible with JSON expected by the Jackson for example?
I mean serialize our way, but deserialize to `Foo` with Jackson.

**Pull request closed without merging**